web/packages/design/src/DataTable/StyledTable.tsx

```diff
@@ -89,6 +90,19 @@ export const StyledTable = styled.table(
   borderRadius
 );
 
+// When `border-collapse: collapse` is set on a table element, Safari incorrectly renders row border with alpha channel.
+// It looks like the collapsed border was rendered twice, that is, opacity 0.07 looks like opacity 0.14 (this is more visible
+// on the dark theme).
+// Sometimes, there is also an artifact visible after hovering the rows - some of them have correct border color, some not.
+// WebKit issue https://bugs.webkit.org/show_bug.cgi?id=35456.
+//
+// `getSolidRowBorderColor` is a workaround. The color is created by lightening or darkening the table background by
+// the value of theme.colors.spotBackground[0].
```

Comment:
No, Firefox and Chrome render the same border color (visually) as before the fix. 
You can try replacing 
```
  tbody tr {
    border-bottom: 1px solid ${getSolidRowBorderColor(props.theme)};
  }
```
with
```
  tbody tr {
    border-bottom: 1px solid ${props.theme.colors.spotBackground[0]};
  }
```
there won't be any difference.

The only thing that has slightly changed is the visibly of borders on hover.
Previously, they were slightly visible:
<img width="437" alt="Screenshot 2023-05-05 at 15 40 45" src="https://user-images.githubusercontent.com/20583051/236473911-be9e4089-c0f8-4ebc-ba19-ae6d577fa808.png">

Now, they are mixed with the hover color:
<img width="437" alt="Screenshot 2023-05-05 at 15 39 46" src="https://user-images.githubusercontent.com/20583051/236473518-e7aa2bb7-5e46-4202-a81e-61b2948d6ba7.png">

I guess that's because previously we had an alpha from border + alpha from hover and they were mixed differently than alpha from hover + solid border color. 

But I don't think it is a problem, they look fine either way.